.circleci/config.yml

```diff
@@ -0,0 +1,149 @@
+  unit_tests:
+    machine: true
+    working_directory: ~/redash
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Install Docker Compose
+          command: |
+            set -x
+            pip install --upgrade pip
+            pip install docker-compose>=1.18
+            docker-compose --version
+      - run:
+          name: Pull images and setup
+          command: |
+            set -x
+            docker-compose up -d
+            sleep 10
+            docker-compose run --rm postgres psql -h postgres -U postgres -c "create database tests;"
+      - run:
+          name: Run tests
+          command: docker-compose run --user root server tests
+      - store_artifacts:
+          path: junit.xml
+      - store_artifacts:
+          path: coverage.xml
+  integration_tests:
+    working_directory: ~/redash
+    machine: true
+    environment:
+      REDASH_SERVER_URL : "http://127.0.0.1:5000/"
+      DOCKER_IMAGE: mozilla/redash-ui-tests
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Install Docker Compose
+          command: |
+            set -x
+            pip install --upgrade pip
+            pip install docker-compose>=1.18
+            docker-compose --version
+      - run:
+          name: Pull redash images
+          command: |
+            set -x
+            docker-compose -f docker-compose.yml up -d
+            sleep 10
+      - run:
+          name: Pull redash-ui-tests
+          command: docker pull "${DOCKER_IMAGE}":latest
+      - run:
+          name: Setup redash instance
+          command: |
+            set -x
+            docker-compose run --rm --user root server create_db
+            docker-compose run --rm postgres psql -h postgres -U postgres -c "create database tests"
+            docker-compose run --rm --user root server /app/manage.py users create_root root@example.com "rootuser" --password "IAMROOT" --org default
+            docker-compose run --rm --user root server /app/manage.py ds new "ui-tests" --type "url" --options '{"url": "uitests"}'
+            docker-compose run -d -p 5000:5000 --user root server
+      - run:
+          name: Run tests
+          command: |
+            set -x
+            docker run --net="host" --env REDASH_SERVER_URL="${REDASH_SERVER_URL}" "${DOCKER_IMAGE}"
+      - store_artifacts:
+          path: report.html
+  build-deploy: # build for master or release branches
+    docker:
+      - image: circleci/node:6
+    working_directory: ~/redash
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Build and deploy
+          command: bin/deploy "$CIRCLE_BRANCH"
+  build-release: # build for tags
+    docker:
+      - image: circleci/node:6
+    working_directory: ~/redash
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Tag deploy
+          command: |
+            set -x
+            ./bin/deploy "$CIRCLE_TAG"
```

Comment:
`build_deploy` and `build_release` are named wrong since the difference between them is really just the docker tag name and aliasing it to `latest`.

Could we in a different step set a new environment variable (e.g. `BUILD_TAG`) depending on which is called? 

So I suggest that there would be two workflow jobs `build_branch` and `build_tag` instead, where the latter requires the first and just creates the alias in addition?